client/components/admin/UserForm.jsx

```diff
@@ -0,0 +1,260 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import toastr from 'toastr';
+import TextInput from '../common/TextInput.jsx';
+import * as userActions from '../../actions/userActions';
+import { addFlashMessage } from '../../actions/flashMessages';
+
+class UserForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      user: {}, // Object.assign({}, props.userValue),
+      errors: {},
+      fullname: Object.assign({}, props.userValue).name,
+      username: Object.assign({}, props.userValue).username,
+      email: Object.assign({}, props.userValue).email,
+      select: Object.assign({}, props.userValue).roleId,
+      password: ''
+    };
+    this.onChange = this.onChange.bind(this);
+    this.updateUser = this.updateUser.bind(this);
+    this.saveUser = this.saveUser.bind(this);
+    this.clearError = this.clearError.bind(this);
+  }
+
+  componentDidMount() {
+    // $('#mySelectBox').on('change', this.updateSelectState);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.userValue.id !== nextProps.userValue.id) {
+      // Necessary to populate form when existing documents loaded directly.
+
+      this.setState({
+        fullname: Object.assign({}, nextProps.userValue).name,
+        username: Object.assign({}, nextProps.userValue).username,
+        email: Object.assign({}, nextProps.userValue).email,
+        select: Object.assign({}, nextProps.userValue).roleId
+      });
+    }
+  }
+
+  onChange(event) {
+    const field = event.target.name;
+    const user = this.state.user;
+    const value = event.target.value;
+    user[field] = event.target.value;
+    if (field === 'name') {
```

Comment:
it is not advisable to use setState several times this way. You can rename the fields to match the state exactly so that you can use `this.setState([field]: value)` instead of the blocks of `if...else`s.